### PR TITLE
Refactor components

### PR DESCRIPTION
**Pull request by martingaston**
Topline changes: 

- Refactor <History> <Examinations> and <Extras> into <CategoryListing> (happy to take suggestions on the name tbh...)
- Add CSS `overflow-wrap` to tile listing
- Modularise <CompleteButton> to work across all current components 
- Styled components :tada: 
- Added page colours to sections
- Added colours to nav bar background on hover
- Redirect `/` route to `/history`
- Fix <ComingSoon> routing
- Tidy up deprecated files

### CODE REVIEW

File: src/components/CategoryListing.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  generateLink(name) {
+    if (this.props.section === "history")
+      return `/${this.props.section}/${name}`;
+    return `/coming-soon`;
+  }
+
+  render() {
+    this.generateLink();
```
Comment (sima-qian) [inline src/components/CategoryListing.js]: Does `this.generateLink()` need to be called here? What's it doing?

Comment (sima-qian) [inline src/components/CategoryListing.js]: Does `this.generateLink()` need to be called here? What's it doing?

File: src/components/TopBar/CompleteButton.js [hunk 2]
```diff
@@ -3,30 +3,22 @@
...
-            }
-          >
-            <img src={tickIcon} />
-          </button>
-        </Link>
+        {link && `<Link to=${link}>`}
+        <button id="complete" data-testid="complete" onClick={() => onClick()}>
+          <img src={tickIcon} />
+        </button>
+        {link && `</Link>`}
```
Comment (sima-qian) [inline src/components/TopBar/CompleteButton.js]: I tested it out, and the complete button still doesn't function properly on the add-station or add-case flow. We can figure it out after this PR, if you like :)

File: src/components/CategoryListing.js (hunk 1 above)
Comment (njons) [reply src/components/CategoryListing.js]: can the following be added to Tile: 
- line-height: 1.2;
- letter-spacing: 1px;
- font-size: 21px;

can one display: flex; be removed

Comment (martingaston) [reply src/components/CategoryListing.js]: Yeah that's a stray, it's in the pound now.

File: src/components/TopBar/CompleteButton.js (hunk 2 above)
Comment (martingaston) [reply src/components/TopBar/CompleteButton.js]: Infurating :scream:

**Pull request merged**